src/java.base/share/classes/jdk/internal/lang/stable/StableValueFactories.java

```diff
@@ -0,0 +1,82 @@
+package jdk.internal.lang.stable;
+
+import jdk.internal.access.SharedSecrets;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+import java.util.function.Supplier;
+
+public final class StableValueFactories {
+
+    private StableValueFactories() {}
+
+    // Factories
+
+    public static <T> StableValueImpl<T> of() {
+        return StableValueImpl.newInstance();
+    }
+
+    public static <T> StableValueImpl<T> of(T value) {
+        final StableValueImpl<T> stableValue = of();
+        stableValue.trySet(value);
+        return stableValue;
+    }
+
+    public static <T> Supplier<T> supplier(Supplier<? extends T> original) {
+        return StableSupplier.of(original);
+    }
+
+    public static <R> IntFunction<R> intFunction(int size,
+                                                 IntFunction<? extends R> original) {
+        return StableIntFunction.of(size, original);
+    }
+
+    public static <T, R> Function<T, R> function(Set<? extends T> inputs,
+                                                 Function<? super T, ? extends R> original) {
+        if (inputs.isEmpty()) {
```

Comment:
If it is worth optimizing the isEmpty scenario, it might be preferable to let each xxxFunction.of return an appropriate instance, to keep the number of varying subclasses to a minimum.

Reply:
I've removed the specialized empty class.